src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -2521,18 +2521,37 @@ pbs_python_populate_svrattrl_from_python_class(PyObject *py_instance,
 	PyObject	*py_resc_hookset_dict0 = (PyObject *)NULL;
 	PyObject	*py_attr_keys = (PyObject *)NULL;
 	PyObject 	*py_val = (PyObject *)NULL;
-	char	*name_str_dup = NULL;
-	char	*val_str_dup = NULL;
+	char		*name_str_dup = NULL;
+	char		*val_str_dup = NULL;
 	int		num_attrs, i;
 	pbs_list_head	svrattrl_list2;
-	int         rc = -1;
+	int		rc = -1;
 	int		hook_set_flag = 0;
 	int		has_resv_duration;
-	char            the_resc[HOOK_BUF_SIZE];
-	char            the_val[HOOK_BUF_SIZE];
+	static char     *the_resc = NULL;
+	static char     *the_val = NULL;
+	static long int resc_buf_size = HOOK_BUF_SIZE;
+	static long int val_buf_size = HOOK_BUF_SIZE;
```

Comment:
Correct me If I am wrong but one could easily see the function is non re-entrant based on the frequent use of global variable log_buffer.  I haven gone through some part of the code, I noticed that it uses strtok_quoted which is a non re-entrant function. I don't think doxygen needs to be modified, since it was a non re-entrant even before this fix was included.